Lib/ntpath.py

```diff
@@ -368,13 +368,18 @@ def expanduser(path):
     If user or $HOME is unknown, do nothing."""
     path = os.fspath(path)
     if isinstance(path, bytes):
+        seps = b'\\/'
         tilde = b'~'
     else:
+        seps = '\\/'
         tilde = '~'
     if not path.startswith(tilde):
         return path
-    i, n = 1, len(path)
-    while i < n and path[i] not in _get_bothseps(path):
+    i = 0
+    for i, char in enumerate(path[1:], start=1):
+        if char in seps:
+            break
+    else:
```

Comment:
@erlend-aasland, this looks a bit better, but I would prefer a nicer approach.

Reply:
Ah, I was under the false impression that `i` was only used in the loop. I see now that it is used later in this function. Sorry, I think it would be better to revert this commit and leave the loop as it is.

Reply:
In general, and when in doubt: always prefer a smaller diff for your PRs. Less churn, easier to review, and a cleaner Git history.

Reply:
Yeah, I'll try to do that as much as possible.